### ISSUE TITLE
Split the activation script into functions or files

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#2134]
Split the Bash activation script into functions and/or files that generally reflect the architecture in https://www.mermaidchart.com/raw/349e994c-c9ec-4dbf-8c74-454872f2a55c?theme=light&version=v0.1&format=svg

This means we should have something like
- start()
- attach()
- invoke()
- attach_in_place()

Although I don't think we need to stick to that strictly. I think a bit of implementer's choice is acceptable here

**Comment by mkenigs** [comment]
Beware of merge conflicts with https://github.com/flox/flox/issues/2130

**Comment by ysndr** [comment]
Can this ticket even be worked on in isolation, considering that the concept of attach is not currently represented in the activate script?

Considering its currently estimated with 1pt i assume this issue does not cover _implementing attaching_?
With that, does it make sense to rephrase this as a refactor of the current capabilities, (invoke, in_place, start) to create a basis to then implement attaching?

**Comment by ysndr** [comment]
**Breakdown**
Refactor _current_ activate structure into at methods that _at least document_ their expected preconditions; i.e. variables to be set.

argument handling and state setup:

https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L108-L355


`command()`

https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L365-L427

interactive()

https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L429-L487


`in_place()`

https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L489-L533

I'd estimate on the upper end of 1 or 2, for understanding and documenting preconditions

**Comment by mkenigs** [comment]
> Can this ticket even be worked on in isolation, considering that the concept of attach is not currently represented in the activate script?

We do in some sense already have attach. `start` would be https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L229-L329 and `attach` would be https://github.com/flox/flox/blob/52df88e4cc2454d52b78d0aa2f183e779da57f0e/assets/activation-scripts/activate#L330-L360

> Edit2: "Attach in Place" is the flox-activation program right? Should this be outside of the "Activation script" box then?

Attach in place will call `flox-activation`, but it will do a few other things as well. But I think you're right we don't need to do that yet for this ticket

**Comment by dcarley** [comment]
I'll raise another PR to apply `shfmt` before closing this.
